tests/components/holiday/test_calendar.py

```diff
@@ -0,0 +1,88 @@
+"""Tests for calendar platform of Holiday integration."""
+
+from datetime import datetime, timedelta
+
+from freezegun.api import FrozenDateTimeFactory
+
+from homeassistant.components.holiday import calendar
+from homeassistant.components.holiday.const import CONF_PROVINCE, DOMAIN
+from homeassistant.const import CONF_COUNTRY
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import entity_registry as er
+from homeassistant.util import dt as dt_util
+
+from tests.common import MockConfigEntry
+
+
+async def assert_setup_calendar(hass: HomeAssistant) -> MockConfigEntry:
+    """Set up the calendar."""
+    config_entry = MockConfigEntry(
+        domain=DOMAIN,
+        data={CONF_COUNTRY: "CH", CONF_PROVINCE: "GE"},
+        title="Switzerland, GE",
+    )
+    config_entry.add_to_hass(hass)
+
+    await hass.config_entries.async_setup(config_entry.entry_id)
+    await hass.async_block_till_done()
+
+    return config_entry
+
+
+async def test_holiday_calendar_entity(
+    hass: HomeAssistant,
+    entity_registry: er.EntityRegistry,
+    freezer: FrozenDateTimeFactory,
+) -> None:
+    """Test HolidayCalendarEntity functionality."""
+    freezer.move_to(datetime(2023, 1, 1, 12, tzinfo=dt_util.UTC))
+
+    config_entry = await assert_setup_calendar(hass)
+    entity = calendar.HolidayCalendarEntity(
```

Comment:
We should not load this directly here but instead make a proper service call to see we get the returned event(s) as expected

Reply:
Can you give me some hint? Both Python and Home Assistant are new to me. Thanks for your patience.

Reply:
https://github.com/home-assistant/core/blob/dev/tests/components/workday/test_binary_sensor.py#L282

Reply:
Thanks, I refactored the test and make a service call now.